src/System.Management.Automation/engine/lang/parserutils.cs

```diff
@@ -785,6 +785,94 @@ internal static object JoinOperator(ExecutionContext context, IScriptExtent erro
             }
         }
 
+        /// <summary>
+        /// The implementation of the PowerShell range operator.
+        /// </summary>
+        /// <param name="context">The execution context in which to evaluate the expression.</param>
+        /// <param name="errorPosition">The position to use for error reporting.</param>
+        /// <param name="lval">The object on which to start.</param>
+        /// <param name="rval">The object on which to stop.</param>
+        /// <returns>The array of objects.</returns>
+        internal static object RangeOperator(ExecutionContext context, IScriptExtent errorPosition, object lval, object rval)
+        {
+            var lbase = PSObject.Base(lval);
+            var rbase = PSObject.Base(rval);
+
+            var lType = lbase.GetType();
+            var rType = rbase.GetType();
+
+            // Return objects of [char] type for explicit [char] cast.
+            // Common case:
+            //     [char]'A'..[char]'Z'
+            // Special case:
+            //     [char]'0'..[char]'9'
+            if (lType == typeof(char) && rType == typeof(char))
+            {
+                return CharOps.Range((char)lbase, (char)rbase);
+            }
+
+            // For arguments of string type and length == 1
+            // in the common case:
+            //     "A".."Z" - return objects of [char] type.
+            //
+            // For arguments of string type and length == 1
+            // in the special case:
+            //     "0".."9" - return objects of [int] type
+            if (lType == typeof(string) && rType == typeof(string))
```

Comment:
It might nice to support:

```
[char]"a" .. "z"
```

It would be less useful to have `"a"..[char]"z"`, but it's not complicated to support both.

You could introduce a local helper function:
```c#
object AsChar(object o)
{
    if (o is char) return o;
    if (o as string s && s.Length == 1 && !Char.IsDigit(s, 0)) return o[0];
    return null;
}

if (AsChar(lbase) as char lc && AsChar(rbase) as char rc)
{
    return CharOps.Range(lc, rc);
}
```

This covers all combos of char/char,string/string, and mixed. I think - browser compiled so may not work :)

Reply:
Thanks for great comment!

Done.